hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapreduce/TestNewCombinerGrouping.java

```diff
@@ -105,7 +106,8 @@ public int compare(Text o1, Text o2) {
 
   @Test
   public void testCombiner() throws Exception {
-    if (!new File(TEST_ROOT_DIR).mkdirs()) {
+    File testDir = new File(TEST_ROOT_DIR);
```

Comment:
best to do the fully delete here, so mkdirs always works, even if test was stopped partway through